File: TextosJapones/src/main/java/org/jisho/textosJapones/database/dao/LegendasDao.java
package org.jisho.textosJapones.database.dao;

import org.jisho.textosJapones.model.entities.FilaSQL;
import org.jisho.textosJapones.model.entities.Processar;
import org.jisho.textosJapones.model.entities.subtitle.Legenda;
import org.jisho.textosJapones.model.exceptions.ExcessaoBd;

import java.util.List;
import java.util.UUID;

public interface LegendasDao {

	List<String> getTabelas() throws ExcessaoBd;

	void createTabela(String base) throws ExcessaoBd;

	void delete(String tabela, Legenda obj) throws ExcessaoBd;
	void insert(String tabela, Legenda obj) throws ExcessaoBd;
	void update(String tabela, Legenda obj) throws ExcessaoBd;

	void comandoUpdate(String update, Processar obj) throws ExcessaoBd;

	List<Processar> comandoSelect(String select) throws ExcessaoBd;

	void comandoDelete(String delete) throws ExcessaoBd;

	void comandoInsert(FilaSQL fila) throws ExcessaoBd;

	void comandoUpdate(FilaSQL fila) throws ExcessaoBd;

	Boolean existFila(String deleteSql) throws ExcessaoBd;

	List<FilaSQL> comandoSelect() throws ExcessaoBd;

}

File: ProcessaTexto/src/main/kotlin/br/com/fenix/processatexto/database/dao/LegendasDao.kt
package br.com.fenix.processatexto.database.dao

import br.com.fenix.processatexto.model.entities.comicinfo.ComicInfo
import br.com.fenix.processatexto.model.entities.processatexto.Processar
import br.com.fenix.processatexto.model.entities.processatexto.Revisar
import br.com.fenix.processatexto.model.entities.subtitle.FilaSQL
import br.com.fenix.processatexto.model.entities.subtitle.Legenda
import java.sql.SQLException
import java.util.*


interface LegendasDao {
    @get:Throws(SQLException::class)
    val tabelas: List<String>

    @Throws(SQLException::class)
    fun createTabela(base: String)

    @Throws(SQLException::class)
    fun delete(tabela: String, obj: Legenda)

    @Throws(SQLException::class)
    fun insert(tabela: String, obj: Legenda)

    @Throws(SQLException::class)
    fun update(tabela: String, obj: Legenda)

    @Throws(SQLException::class)
    fun select(tabela: String, id: UUID) : Optional<Legenda>

    @Throws(SQLException::class)
    fun comandoUpdate(update: String, obj: Processar)

    @Throws(SQLException::class)
    fun comandoSelect(select: String): MutableList<Processar>

    @Throws(SQLException::class)
    fun comandoDelete(delete: String)

    @Throws(SQLException::class)
    fun comandoInsert(fila: FilaSQL)

    @Throws(SQLException::class)
    fun comandoUpdate(fila: FilaSQL)

    @Throws(SQLException::class)
    fun existFila(deleteSql: String): Boolean

    @Throws(SQLException::class)
    fun comandoSelect(): MutableList<FilaSQL>
}